cloudinit/net/dhcp.py

```diff
@@ -679,6 +703,38 @@ def dhcp_discovery(
             )
             raise NoDHCPLeaseError from error
 
+    @staticmethod
+    def parse_unknown_options_from_packet(
+        data: bytes, number: int
+    ) -> Optional[bytes]:
+        """get a specific option from a binary lease file
+
+        @param data: Binary lease data
+        @param number: Option number to return
+        @return: the option (bytes) or None
```

Comment:
Let's add a comment linking to your upstream dhcpcd issue here that this function is only needed temporarily until upstream dhcpcd supports option 245 https://github.com/NetworkConfiguration/dhcpcd/issues/282